array_family/flex_ext.py

```diff
@@ -645,8 +645,8 @@ def match(
         ann = AnnAdaptorSelfInclude(r, 3)
         ann.query(x)
 
-        mm = cctbx.array_family.flex.int(range(xyz.size()))
-        nn, distance = ann.nn, cctbx.array_family.flex.sqrt(ann.distances)
+        mm = cctbx.array_family.flex.size_t_range(xyz.size())
+        nn, distance = ann.nn.as_size_t(), cctbx.array_family.flex.sqrt(ann.distances)
```

Comment:
Thanks, makes much more sense - I am not sure why I did not do this before, but it is possible (as mentioned by @ndevenish ) that `flex.size_t` was not a thing when I added the original implementation